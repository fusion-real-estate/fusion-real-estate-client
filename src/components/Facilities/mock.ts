export default {
  title: 'Facilities',
  airConditioning: 'Air Conditioning',
  barbeque: 'Barbeque'
}
