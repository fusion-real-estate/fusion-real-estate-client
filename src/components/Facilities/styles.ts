import styled, { css } from 'styled-components'

export const Wrapper = styled.main``

export const ListIcons = styled.div`
  display: flex;
  align-items: center;
`

export const IconWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    padding: 1rem;

    background: ${theme.colors.lightGray};

    border-radius: 50%;

    margin-right: ${theme.spacings.xsmall};
  `}
`

export const BoxList = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    margin-top: ${theme.spacings.xxsmall};
    margin-right: ${theme.spacings.xxsmall};

    svg {
      height: 2rem;
      width: 2rem;
      color: ${theme.colors.gray};
    }
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-weight: ${theme.font.normal};
    font-size: ${theme.font.sizes.xsmall};
  `}
`
