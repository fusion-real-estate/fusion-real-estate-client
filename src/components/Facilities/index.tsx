import Heading from 'components/Heading'

import Icon from '@mdi/react'

import { mdiAirFilter, mdiGrillOutline } from '@mdi/js'

import * as S from './styles'

export type FacilitiesProps = {
  title: string
  airConditioning?: string
  barbeque?: string
}

const Facilities = ({ title, airConditioning, barbeque }: FacilitiesProps) => (
  <S.Wrapper>
    <Heading lineLeft>{title}</Heading>
    <S.ListIcons>
      <S.BoxList>
        <S.IconWrapper>
          <Icon path={mdiAirFilter} />
        </S.IconWrapper>
        <S.Description>{airConditioning}</S.Description>
      </S.BoxList>
      <S.BoxList>
        <S.IconWrapper>
          <Icon path={mdiGrillOutline} />
        </S.IconWrapper>
        <S.Description>{barbeque}</S.Description>
      </S.BoxList>
    </S.ListIcons>
  </S.Wrapper>
)

export default Facilities
