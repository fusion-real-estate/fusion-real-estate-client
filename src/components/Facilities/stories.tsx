import { Story, Meta } from '@storybook/react/types-6-0'
import Facilities, { FacilitiesProps } from '.'

import mockFacilities from './mock'

export default {
  title: 'Facilities',
  component: Facilities,
  args: mockFacilities
} as Meta

export const Basic: Story<FacilitiesProps> = (args) => <Facilities {...args} />
