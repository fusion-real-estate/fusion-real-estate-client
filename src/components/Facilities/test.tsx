import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Facilities from '.'

const props = {
  icon:
    'https://res.cloudinary.com/dtykejdjn/image/upload/v1651790246/air_filter_78f05f422a.png',
  name: 'Ar Condicionado'
}

describe('<Facilities />', () => {
  it('should render the heading', () => {
    renderWithTheme(<Facilities {...props} />)

    expect(screen.getByRole('heading', { name: props.name }))
  })
})
