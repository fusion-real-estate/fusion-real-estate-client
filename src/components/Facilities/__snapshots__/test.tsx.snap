// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Facilities /> should render the heading 1`] = `
.c0 {
  color: #373737;
  font-size: 2.0rem;
  padding-left: 0.8rem;
  border-left: 0.7rem solid #34ABD0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1.6rem;
  margin-right: 4.0rem;
}

.c2 svg {
  height: 2rem;
  width: 2rem;
  color: #7D7D7D;
}

.c3 {
  color: #7D7D7D;
  font-weight: 400;
  font-size: 1.2rem;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.8rem;
  }
}

<main
  class=""
>
  <h2
    class="c0"
    color="black"
  >
    Facilities
  </h2>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3"
      />
    </div>
  </div>
</main>
`;
